src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -54,34 +54,39 @@ static void HandleNodeResolve(void * context, DnssdService * result, CHIP_ERROR
 
     DiscoveredNodeData nodeData;
 
-    Platform::CopyString(nodeData.hostName, result->mHostName);
-    Platform::CopyString(nodeData.instanceName, result->mName);
+    Platform::CopyString(nodeData.hostName, results.data()[0].mHostName);
+    Platform::CopyString(nodeData.instanceName, results.data()[0].mName);
+    nodeData.port = results.data()[0].mPort;
 
-    if (result->mAddress.HasValue())
+    size_t addressesFound = 0;
+    for (auto & result : results)
     {
-        nodeData.ipAddress[0] = result->mAddress.Value();
-        nodeData.interfaceId  = result->mInterface;
-        nodeData.numIPs       = 1;
-    }
-    else
-    {
-        nodeData.numIPs = 0;
-    }
-
-    nodeData.port = result->mPort;
+        if (result.mAddress.HasValue())
+        {
+            if (addressesFound == ArraySize(nodeData.ipAddress))
+            {
+                // Out of space.
+                break;
+            }
+            nodeData.ipAddress[addressesFound] = result.mAddress.Value();
+            nodeData.interfaceId               = result.mInterface;
+            ++addressesFound;
+        }
 
-    for (size_t i = 0; i < result->mTextEntrySize; ++i)
-    {
-        ByteSpan key(reinterpret_cast<const uint8_t *>(result->mTextEntries[i].mKey), strlen(result->mTextEntries[i].mKey));
-        ByteSpan val(result->mTextEntries[i].mData, result->mTextEntries[i].mDataSize);
-        FillNodeDataFromTxt(key, val, nodeData);
+        for (size_t i = 0; i < result.mTextEntrySize; ++i)
```

Comment:
If I get multiple results, does this mean all the text entries are duplicated?

Reply:
Again, for the Darwin impl these text entries are identical for all the structs coming in.

Reply:
I think I misread how fillNodeData works - I assume 'last writer wins'? If so this sounds correct as a union but I would add a comment of why we think this is ok (I don't think CHIP has any business in having per-interface or per-record specific text entries).

Reply:
Done by changing the API so this is a non-issue.